docs/reference/fields.txt

```diff
@@ -1386,8 +1386,39 @@ requiring the usage of the aggregation pipeline for both queries and updates,
 applications should avoid using dots in field names and starting field names
 with the dollar sign if possible.
 
-The Ruby driver `currently prohibits
-<https://jira.mongodb.org/browse/RUBY-2528>`_ inserting documents whose
-field names contain dots or begin with the dollar sign. However, if such
-documents are inserted using other software, Mongoid and the Ruby driver
-provide limited support for retrieving and operating on these documents.
+Mongoid, starting in version 8, now allows users to access fields that begin with
+dollar signs and that contain dots/periods. They can be accessed using the ``send``
+method as follows:
+
+.. code::
+
+    class User
+      include Mongoid::Document
+      field :"first.last", type: String
+      field :"$_amount", type: Integer
+    end
+
+    user = User.first
+    user.send(:"first.last")
+    # => Mike.Trout
+    user.send(:"$_amount")
```

Comment:
Defining Ruby methods which do not adhere to [Ruby's naming conventions](https://docs.ruby-lang.org/en/2.4.0/syntax/methods_rdoc.html#label-Method+Names) and require using the Kernel `.send` is an anti-pattern/code smell. I have discussed many times that we should not abuse Ruby's Kernel conventions, and this case is no different.

Instead:
- Do not create getters/setters if they do not adhere to Ruby's naming convention.
- Define an alternate Mongoid method to access these, for example `#read_attribute` which is already in existence may work, or maybe we add a new method like `#get_attribute`.

Reply:
@johnnyshields We already discussed your position on `send` etc. in the previous dots/dollars PRs, please do not bring this up in every PR going forward. The `read_attribute` addition I agree with.